src/core/DragControls.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react'
+import { ReactThreeFiber, useThree, useFrame } from '@react-three/fiber'
+import { DragControls as DragControlsImpl } from 'three-stdlib'
+import { Group, Object3D } from 'three'
+
+export type DragControls = ReactThreeFiber.Overwrite<
+  ReactThreeFiber.Object3DNode<DragControlsImpl, typeof DragControlsImpl>,
+  {
+    camera?: THREE.Camera
+    objects: Object3D[]
+  }
+>
+
+export const DragControls = React.forwardRef<DragControlsImpl, DragControls>(
+  ({ objects, camera, children, ...restProps }, ref) => {
+    const invalidate = useThree(({ invalidate }) => invalidate)
+    const defaultCamera = useThree(({ camera }) => camera)
+    const gl = useThree(({ gl }) => gl)
+
+    const explCamera = camera || defaultCamera
+    const [controls] = React.useState<DragControlsImpl>(() => new DragControlsImpl(objects, explCamera, gl.domElement))
```

Comment:
So you have to attach dragControls to the meshs your want to use?
It might make more sense as a HOC to be consistent with the other Control APIs we have.

Reply:
I considered using a HOC. The problem is that one can't attach/detach meshes after instantation like one can with `TransformControls`, as all mesh instances have to be passed to the constructor. So forwarding a controls `ref` to the parent component can only happen once all child meshes have rendered.

But yeah you're right HOC is probably better. I will try make it work.

Reply:
Is there a downside to re-making the the `DragControl` impl when new children are added?

Reply:
I tried that but it causes all event listeners added by the parent component to no longer work. Perhaps I could store event callbacks in the HOC and simply re-apply them to the new `controls` instance. (This would have to be done anyway given that some children might be rendered conditionally)